scripts/mbedtls_dev/crypto_reference.py

```diff
@@ -236,3 +237,74 @@ def output_bytes(self, length: int) -> bytes:
         self.offset += length
         return output
 
+    def output_ecc_weierstrass(self, order: int) -> bytes:
+        """Return an ECC private key on a curve of the given order.
+
+        Follow the derivation method specified in the PSA Cryptography API
```

Comment:
The order of operations is slightly different here (masking high bits is done at the end instead of after drawing the string), this doesn't seem significant though so assume this is OK and this comment can stand.

Reply:
I don't understand where you see a difference: both the code and the spec say
1. draw (`output_bytes`)
2. convert to `int`
3. mask
4. test range

Reply:
I'm looking at https://armmbed.github.io/mbed-crypto/html/api/ops/kdf.html?highlight=psa_key_derivation_output_key#c.psa_key_derivation_output_key which says:

1. Draw a byte string of length ceiling(m/8) bytes.
2. If m is not a multiple of 8, set the most significant (8 * ceiling(m/8) - m) bits of the first byte in the string to zero. 
3. Convert the string to integer k by decoding it as a big-endian byte string.
4. If k > N - 2, discard the result and return to step 1.
5. Output k + 1 as the private key.

i.e., steps 2 and 3 the other way around. But it seems equivalent so not an issue unless you want to be extra-precise about following the spec.

Reply:
Is this relevant? As far I understand it does not matter if we clear these bits in the byte sting (first byte in the string) or after conversion to int.

```
Let's say we want to clear 3 bits
Byte array: 0xAA 0xBB 0xCC 0xDD       ; After clearing 3 most significant bits: 0x0A 0xBB 0xCC 0xDD
Int value: 0xAABBCCDD                 ; After clearing 3 most significant bits: 0x0ABBCCDD
```

Reply:
Ah, right, indeed, the python code is the other way round. But, indeed, it doesn't matter. It would be inconvenient to do the masking on the byte string because python requires converting the bytes to integers for masking anyway. I'll add a comment to explain how the code matches the spec.